### ISSUE TITLE
how to change default python version in uv? 

### ISSUE THREAD

**Issue by aizimuji** [astral-sh/uv#8135]
for example, i just install python 3.13

if i run `uv run python` , python 3.13 version is used 

How can i change this default version to other 

maybe it shoud add a command like uv python setdeault 3.12 ?

**Comment by charliermarsh** [comment]
You can add `.python-version` file to the current directory (with `3.13`) and we'll respect it!

**Comment by leodevian** [comment]
I set the `UV_PYTHON` environment variable to `3.12` and uv now defaults to Python 3.12.

**Comment by zanieb** [comment]
> You can add .python-version file to the current directory (with 3.13) and we'll respect it!

And you do this with `uv python pin 3.13`

I plan on adding the ability to change the default globally in the next breaking version — i.e. after https://github.com/astral-sh/uv/pull/6370 is merged I'll add a `--global` flag to `uv python pin`.

**Comment by aizimuji** [comment]
> You can add `.python-version` file to the current directory (with `3.13`) and we'll respect it!

sometimes I need to run python interpreter for a specific version , not in a project folder 

for now I need to run uv run -p 3.12 python to run python repl in 3.12 version 

and many command need to attach -p 3.12 to sprcify the python version

so it's very useful to specify a global default python version

**Comment by sb-travelperk** [comment]
> > You can add .python-version file to the current directory (with 3.13) and we'll respect it!
> 
> And you do this with `uv python pin 3.13`
> 
> I plan on adding the ability to change the default globally in the next breaking version — i.e. after #6370 is merged I'll add a `--global` flag to `uv python pin`.

Any update on this @zanieb?   #6370 has been merged in and I was expecting to see something new in v0.5. But the changelog doesn't contain any mention of a global flag or default version. 🤔

**Comment by kevinrenskers** [comment]
When I run `uv run python` in my home folder, it starts Python 3.11.10. Where is this default version coming from? There is no `.python-version` file in my home folder. Newer Python versions are available, so I don't understand why it picks 3.11?

**Comment by the-citto** [comment]
I found this issue still open while searching for setting up `UV` defaults 

it would be nice to have more (most?) `CLI` settings configurable, perhaps in the `uv.toml`?

specifically for the OP issue, the default python version for `UV` to use - with the higher-precedence files or the `CLI` able to override the default(s)

#

one of the things I was hoping for was being able to automate project's initialization - searching for `requires-python` I stumbled here, but also for `ruff` 😉, `mypy`, `pyright`, and several more settings

I naively tried creating a `$XDG_CONFIG_HOME/uv/pyproject.toml` - of course no joy with that, since 

> User-and system-level configuration must use the `uv.toml` format, rather than the `pyproject.toml` format, as a `pyproject.toml` is intended to define a Python project. [1](https://docs.astral.sh/uv/configuration/files/)

I also tried `dependency-metadata` -> `requires-python` in `$XDG_CONFIG_HOME/uv/uv.toml` but no luck there either

_Note_: I just installed `UV`, tired of having to continuously having to set up the full `pyenv + git + venv + pip + pip-tools + ... + setuptools` chain all the times, so please forgive if something will come out too naive, and if something like this has already been discussed

**Comment by trim21** [comment]
👀 so curretnly no way to set default python version for `uv tool install` ?

**Comment by Hritik14** [comment]
btw if you're using ``pyenv`` you can do ``pyenv global VERSION``

**Comment by zanieb** [comment]
cc @jtfmumm

**Comment by dantetemplar** [comment]
@zanieb so it will be --global flag in near future?

**Comment by zanieb** [comment]
`--global` is already available, @jtfmumm it doesn't look like it's used for tool installs though? I have a vague recollection of discussion about that?

**Comment by jtfmumm** [comment]
Closing in favor of #12921.
